### PR TITLE
Updated IntegrationTestHelper to mock all api functions

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
N/A

#### What's this PR do?
Refactors `IntegrationTestHelper` to stub out all the API methods.

#### How should this be manually tested?
Test should pass

### CODE REVIEW

File: static/js/util/integration_test_helper.js [hunk 1]
```diff
@@ -36,12 +36,16 @@ export default class IntegrationTestHelper {
...
 
-    this.getChannelStub = this.sandbox.stub(api, "getChannel")
-    this.getChannelStub.throws(new Error("not implemented"))
-    this.createPostStub = this.sandbox.stub(api, "createPost")
-    this.createPostStub.throws(new Error("not implemented"))
-    this.createCommentStub = this.sandbox.stub(api, "createComment")
-    this.createCommentStub.throws(new Error("not implemented"))
+    // stub all the api functions
+    // NOTE: due to a regression in sinon 2.x, if you use `callsFake` you must
+    //       call `resetBehavior()` on the stub first or the error still raises
```
Comment (noisecapella) [inline static/js/util/integration_test_helper.js]: I checked and it looks like this is fixed in sinon 3.x FYI

**Comment by noisecapella** [comment]
Looks like there are some errors in the PR tests

File: static/js/util/integration_test_helper.js (hunk 1 above)
Comment (rhysyngsun) [inline static/js/util/integration_test_helper.js]: I tried upgrading and these still fail when I remove the `resetBehavior` calls

**Comment by rhysyngsun** [comment]
PR tests were failing because this commit added more hardcoded stubs: https://github.com/mitodl/open-discussions/commit/5ad04499d791d31a35312e6d4ec57189665c9209#diff-fd6fb1b588e6057629caba91f07d6dbbR46

I just rebased and updated the one commit to remove those as well. I'll merge when tests pass again.

**Pull request merged**